php-checks/src/main/java/org/sonar/php/checks/security/AuthorizationsCheck.java

```diff
@@ -0,0 +1,198 @@
+      if (isLaravelGateMethod(memberAccessTree, "define")) {
+        argument = argument(tree, "callback", 1);
+      }
+      if (isLaravelGateMethod(memberAccessTree, "before") || isLaravelGateMethod(memberAccessTree, "after")) {
+        argument = argument(tree, "callback", 0);
+      }
+      if (argument.isPresent()) {
+        ExpressionTree callArgumentTree = argument.get().value();
+        if (callArgumentTree.is(FUNCTION_EXPRESSION)) {
+          checkReturnStatements((FunctionExpressionTree) callArgumentTree, LARAVEL_GATE_CLOSURE_COMPLIANT_RETURN_VALUES::contains);
+        }
+      }
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private boolean isLaravelGateMethod(MemberAccessTree memberAccessTree, String expectedMethod) {
+    ExpressionTree receiver = memberAccessTree.object();
+    Tree method = memberAccessTree.member();
+    return method.is(Kind.NAME_IDENTIFIER) &&
+      ((NameIdentifierTree) method).text().equals(expectedMethod) &&
+      receiver.is(Kind.NAMESPACE_NAME) &&
+      getFullyQualifiedName((NamespaceNameTree) receiver).toString().equalsIgnoreCase(LARAVEL_GATE_NAMESPACE);
+  }
+
+  private static boolean isMethodInheritedFromClassOrInterface(String qualifiedName, MethodDeclarationTree methodDeclarationTree) {
+    ClassDeclarationTree classTree = (ClassDeclarationTree) findAncestorWithKind(methodDeclarationTree, singletonList(Kind.CLASS_DECLARATION));
+    if (classTree != null) {
+      return get(classTree).isSubTypeOf(qualifiedName(qualifiedName)).isTrue();
+    }
+    return false;
+  }
+
+  private void checkReturnStatements(FunctionTree methodDeclarationTree, Predicate<String> predicate) {
+    List<ReturnStatementTree> returnStatements = descendants(methodDeclarationTree, ReturnStatementTree.class).collect(Collectors.toList());
+    boolean compliant = false;
+    for (ReturnStatementTree returnStatementTree : returnStatements) {
+      ExpressionTree returnExpressionTree = returnStatementTree.expression();
+      if (returnExpressionTree.is(NUMERIC_LITERAL, REGULAR_STRING_LITERAL)) {
+        compliant = false;
+      } else if (returnExpressionTree.is(FUNCTION_CALL)) {
+        compliant = isFunctionCallCompliant((FunctionCallTree) returnExpressionTree);
+      } else if (returnExpressionTree.is(VARIABLE_IDENTIFIER)) {
+        compliant = isVariableValueCompliant((VariableIdentifierTree) returnExpressionTree, predicate);
+      } else if (returnExpressionTree.is(CLASS_MEMBER_ACCESS)) {
+        compliant = isMemberValueCompliant((MemberAccessTree) returnExpressionTree, predicate);
+      } else if (returnExpressionTree.is(NULL_LITERAL)) {
+        compliant = isNullValueCompliant((LiteralTree) returnExpressionTree, predicate);
+      } else if (returnExpressionTree.is(BOOLEAN_LITERAL)) {
+        compliant = isBooleanValueCompliant((LiteralTree) returnExpressionTree, predicate);
+      } else {
+        compliant = true;
+      }
+      if (compliant) {
+        return;
+      }
```

Comment:
Just an idea, move this logic with all their methods into an inner class like CompliantReturnStatement. On which you can call isCompliant.